### PR TITLE
contributing and code of conduct

### PR DESCRIPTION
**Pull request by jackjt8**

### CODE REVIEW

**Comment by jackjt8** [comment]
Just need updated build instructions and whatnot... @NotStirred

File: CONTRIBUTING.md
Comment (NotStirred) [inline CONTRIBUTING.md:47]: #### Make (more significant) changes locally

Intellij Community Edition is recommended, get it [here](https://www.jetbrains.com/toolbox-app/)
 - You'll want to download JetBrains Toolbox, and download `Intellij IDEA Community Edition` from there

##### Fork the project to your account
Go to https://github.com/chunky-dev/chunky, click on `Fork` (usually in the top right)

##### Clone the project.
with Intellij: `File` -> `New` -> `Project from Version Control`
- In the url put `https://github.com/GITHUB_USERNAME/chunky`
- You can also select a location to clone to here.

with Github desktop: `Clone a repository` -> select `URL`, paste `https://github.com/GITHUB_USERNAME/chunky`

with CLI: `git clone git@github.com:GITHUB_USERNAME/chunky.git` or (outdated) `git clone https://github.com/GITHUB_USERNAME/chunky`

##### Open the Project
From inside Intellij (if you cloned with intellij you can likely skip this step)
- `File` -> `Open` -> Select the first `build.gradle` file within the cloned folder

You may have to wait some time while intellij indexes the project, and imports it. (The little loading bar in the bottom right) 

##### Select the correct JDK version to use to build the project
Oracle Java 8:
- `File` -> `Project Structure` -> `Project` (should be the default)
  - If intellij detected your install: select the `SDK` dropdown, select the oracle 8 jdk install
  - If not: click `Edit`, then the little `+` at the top, `Add JDK` and find your oracle 8 jdk install location (select the main jdk folder).
    - Go back to the `Project` tab, and select the newly added jdk.

TODO: ADD INFO ABOUT NULLABLE AND NOTNULL ANNOTATIONS (adding them to compiler options)

##### Create a new branch for your change
In the bottom right of Intellij, there is a fork button, by default it will say `master`, select it, at the top select `New Branch`, and name it after what you're intending to fix. (ideally something very short, no more than a few words). eg: `fix_scene_tab_typo` or `fix-scene-tab-typo`  is fine.
 - Make sure the `Checkout branch` box is checked when creating a new branch (to actually change to it once it's created)

##### Make your change
Some useful Intellij keybinds to help you navigate around:
 - `Ctrl-n`, Find a file (class) by name

##### Commit your changes
After making your changes, select the `Commit` button (looks like a green tick) in the top right
Select all of the changes you want to make, and add a description of what you changed.

Comment (jackjt8) [reply CONTRIBUTING.md:47]: What about JDK 9+? Should we attach a_Tom's patch or exclude it?

Comment (NotStirred) [reply CONTRIBUTING.md:47]: I think it's best to wait for #1110 to be merged, but up to you

Comment (jackjt8) [reply CONTRIBUTING.md]: Noted. I'll wait for #1110 unless people ask for it.

**Comment by jackjt8** [comment]
Well know I know how to do cherry picks...

File: CONTRIBUTING.md
Comment (leMaik) [inline CONTRIBUTING.md:99]:
```suggestion
Once your PR is merged, your contributions will be publicly available in the next Chunky snapshot and eventually a stable release.
```

File: CONTRIBUTING.md
Comment (leMaik) [reply CONTRIBUTING.md]: Now that #1110 is finally merged, are there any changes required to this PR? :thinking: I had to set my project sdk to Java *17* (not 8) in order to get IntelliJ to work. A small hint about requiring the Java 17 SDK (e.g. openjdk-17-jdk on Ubuntu) would also be nice.

File: CONTRIBUTING.md
Comment (ShirleyNekoDev) [inline CONTRIBUTING.md:66]: IntelliJ capitalization ;)

Comment (jackjt8) [reply CONTRIBUTING.md:66]: crtl+f failed me

File: CONTRIBUTING.md
Comment (leMaik) [inline CONTRIBUTING.md:9]:
```suggestion
When contributing to this repository, please consider discussing significant changes you wish to make via an issue or the [#tech](https://discord.com/channels/541221265512464394/545374333883777037) channel on our [Discord server][chunky-discord] prior to making a change. Correcting spelling or grammar mistakes, or fixing minor bugs are not considered significant. A full rewrite, adding new features, etc. can be considered more significant changes.
```

File: CONTRIBUTING.md
Comment (leMaik) [inline CONTRIBUTING.md:31]: Maybe we should mention the _good first issue_ label and add a link to filter by that?

File: CONTRIBUTING.md
Comment (leMaik) [inline CONTRIBUTING.md:73]: Isn't that button a _branch_ button then?

File: CONTRIBUTING.md
Comment (leMaik) [inline CONTRIBUTING.md:78]:
```suggestion
 - `Ctrl + N`, Find a file (class) by name
```

File: CONTRIBUTING.md
Comment (leMaik) [inline CONTRIBUTING.md:85]:
```suggestion
Prior to making a Pull Request please test your changes. Within IntelliJ you can run chunky\src\java\se.llbit\chunky\main\Chunky.java:line64 assuming you have got IntelliJ setup correctly. At the time of writing, Chunky's gradlew version is 7.4.
```

File: CONTRIBUTING.md
Comment (leMaik) [inline CONTRIBUTING.md:93]: A PR to the docs repoisotry?

File: CONTRIBUTING.md
Comment (leMaik) [inline CONTRIBUTING.md:97]:
```suggestion

If you get stuck at any point along your PR journey, do not hesitate to ask for help or opinions on Discord or in the PR's discussion. We appreciate your contribution and want it to be merged as much as you do. 😉 

```

File: CONTRIBUTING.md
Comment (jackjt8) [reply CONTRIBUTING.md]: Fixed; Might need to add a few screenshots so people know what button to press. Though that could be done later if needed.

File: CONTRIBUTING.md
Comment (jackjt8) [reply CONTRIBUTING.md]: The links direct people to GitHub documentation on how to link issues and PRs, in addition to certain checkboxes that are recommended.

I fixed line 93 to remove the `Docs` confusion.

**Comment by leMaik** [comment]
Is this ready for merge? We should add a link to the instructions to the _Hacking on Chunky_ readme section

**Comment by jackjt8** [comment]
> Is this ready for merge? We should add a link to the instructions to the _Hacking on Chunky_ readme section

Some of the of the stuff in _Hacking on Chunky_ makes more sense to be within _contributing_ so I've added it. Should we retain this information within the readme OR remove the duplication? Both cases will get a link added.

There is one remaining TODO but it doesn't impact too much:
```TODO: ADD INFO ABOUT NULLABLE AND NOTNULL ANNOTATIONS (adding them to compiler options)```

File: CONTRIBUTING.md
Comment (NotStirred) [inline CONTRIBUTING.md:80]:
```suggestion
Add `se.llbit.annotation.Nullable` and `se.llbit.annotation.NotNull` to the compiler options of Intellij for IntelliSense recommendations
- `File` -> `Settings` -> `Build, Execution, Deployment` -> `Compiler` (select it, not the dropdown)
  - Click `Configure annotations...`
    - In the `Nullable` tab add `se.llbit.annotation.Nullable`
    - In the `NotNull` tab add `se.llbit.annotation.NotNull`
    - Intellij must be restarted to update IntelliSense
```

Comment (NotStirred) [reply CONTRIBUTING.md:80]: It would be a good idea to check this looks okay before merging, as I wrote it in a text editor

Comment (jackjt8) [reply CONTRIBUTING.md:80]: Unable to find `se.llbit.annotation.Nullable`. It's recommending to use `NotNull (se.llbit.util)` instead.

Unable to find `se.llbit.annotation.NotNull`. It's recommending to use `NotNull (se.llbit.util)` instead.

Comment (NotStirred) [reply CONTRIBUTING.md:80]: oh, on latest they are in `se.llbit.util.annotation`, apologies

Comment (jackjt8) [reply CONTRIBUTING.md:80]: Given I am just adding two files a rebase isn't needed but I can test on `master` to check these.

**Comment by jackjt8** [comment]
Ready

**Review by leMaik — APPROVED** [review]

**Pull request merged**
